src/components/organisms/course-overview/course-overview.tsx

```diff
@@ -76,11 +77,15 @@ export const CourseOverview: FC<CourseOverviewProps> = ({
         color="warning"
       />
 
-      <Button className={styles.courseEnroll} onClick={handleEnroll}>
+      <Button
+        className={styles.courseEnroll}
+        onClick={handleEnroll}
+        disabled={!canStudy}
+      >
         {buttonText}
       </Button>
 
-      {isDevEnv && (
+      {isEnrolled && (
```

Comment:
```suggestion
      {isDevEnv && isEnrolled && (
```

Функционал пока недоступен в продовом окружении, используется в дев-режиме для отладки